wgpu-native/src/device.rs

```diff
@@ -310,30 +320,33 @@ impl DestroyedResources<back::Backend> {
 
 fn map_buffer(
     raw: &<back::Backend as hal::Backend>::Device,
+    limits: &hal::Limits,
     buffer: &mut resource::Buffer<back::Backend>,
-    range: std::ops::Range<u64>,
-    read: bool,
-    write: bool,
+    mut range: std::ops::Range<u64>,
```

Comment:
It seems like we could accept `&Range<u64>` and clone/make a new range internally, because most callers appear to `clone` anyway